### PR TITLE
Improve TryReadStream with simplification & fix of Stream Invalid Length cutting off Streams

### PR DESCRIPTION
**Pull request by sbruyere**
- Fix of Stream invalid Length issue causing stream data being cut off: fix https://github.com/UglyToad/PdfPig/issues/809

- Improve Stream Token read performance by:
  -  simplifying TryReadStream(), avoiding use of MemoryStream, with benefice of already existing Memory Span of "inputBytes"
  - removing the unecessary List<>

### CODE REVIEW

**Comment by BobLd** [comment]
@sbruyere thanks a lot for that, I'm reviewing the changes. In the meantime, can you add unit tests to cover your changes?

**Comment by sbruyere** [comment]
> @sbruyere thanks a lot for that, I'm reviewing the changes. In the meantime, can you add unit tests to cover your changes?

Done ! Tell me if it's ok for you. Thanks for the review. :)

File: src/UglyToad.PdfPig/Tokenization/Scanner/PdfTokenScanner.cs [hunk 1]
```diff
@@ -330,150 +331,129 @@ private bool TryReadStream(long startStreamTokenOffset, bool getLength, [NotNull
...
+            inputBytes.Seek(streamDataStart);
+            inputBytes.Read(data);
 
-                    memoryStream.Read(data, 0, (int)dataLength);
+            inputBytes.Seek(streamDataEnd);
 
-                    stream = new StreamToken(streamDictionaryToken, data);
-                }
-            }
+            stream = new StreamToken(streamDictionaryToken, data.ToArray());
```
Comment (BobLd) [inline src/UglyToad.PdfPig/Tokenization/Scanner/PdfTokenScanner.cs:456]: can we avoid the `ToArray()` alloc by doing the following:
```csharp
            Memory<byte> data = new byte[dataLength];

            inputBytes.Seek(streamDataStart);
            inputBytes.Read(data.Span);

            inputBytes.Seek(streamDataEnd);

            stream = new StreamToken(streamDictionaryToken, data);
```

**Review by BobLd — CHANGES_REQUESTED** [review]

**Comment by BobLd** [comment]
I've finally managed to findtime to review. I've added 1 code change request. If you have time, also add a comment to explain we're not using the lenght anymore

File: src/UglyToad.PdfPig/Tokenization/Scanner/PdfTokenScanner.cs (hunk 1 above)
Comment (sbruyere) [reply src/UglyToad.PdfPig/Tokenization/Scanner/PdfTokenScanner.cs]: Thanks, far better, adressed with : https://github.com/UglyToad/PdfPig/pull/838/commits/ea82bf8aa539b87da8edcb3508f1f7a80998ab35

**Review by BobLd — APPROVED** [review]

**Pull request merged**